rcl/test/rcl/test_events.cpp

```diff
@@ -121,17 +115,13 @@ class CLASSNAME (TestEventFixture, RMW_IMPLEMENTATION) : public ::testing::Test
 
   void setup_publisher_and_subscriber(
     const rcl_publisher_event_type_t & pub_event_type,
-    const rcl_subscription_event_type_t & sub_event_type)
+    const rmw_qos_profile_t pub_qos_profile,
```

Comment:
Modified this function to take in `qos_profile` for publisher and subscriptions separately. Moreover, this function no longer asserts discovery, that is moved into a new function `setup_publisher_and_subscriber_and_assert_discovery`